climada/hazard/storm_europe.py

```diff
@@ -199,6 +206,209 @@ def _read_one_nc(self, file_name, centroids):
+                           for date_i, ens_i in zip(date_to_str(self.date),
+                                                    stacked2.epsd_1.values+1)
+                           ]
+        self.frequency = np.divide(
+                np.ones_like(self.event_id),
+                ncdf.epsd_1.size)
+        if not description:
+            description = (model_name +
+                           ' weather forecast windfield ' +
+                           'for run startet at ' +
+                           run_date.strftime('%Y%m%d%H'))
+
+        self.tag = TagHazard(
+                HAZ_TYPE, 'Hazard set not saved, too large to pickle',
+                description=description
+            )
+        # close netcdf file
+        ncdf.close()
+        self.check()
+
+    def read_icon_grib(self, run_date, event_date=None,
+                       model_name='icon-eu-eps', description=None,
+                       grib_dir=None, delete_raw_data=True):
+        """Clear instance and download and read dwd icon weather forecast
+        footprints into it. One event is one full day in UCT. Current setup
+        works for runs starting at 00H and 12H. Otherwise the aggregation is
+        inaccurate, because of the given file structure with 1-hour, 3-hour
+        and 6-hour maxima provided.
+
+        Parameters:
+            run_date (datetime): The starting timepoint of the forecast run
+                of the icon model
+            event_date (datetime, optional): one day within the forecast
+                period, only this day (00H-24H) will be included in the hazard
+            model_name (str,optional): select the name of the icon model to
+                be downloaded. Must match the url string
+                (see _download_icon_grib for further info)
+            description (str, optional): description of the events, defaults
+                to 'icon weather forecast'
+            grib_dir (str, optional): path to folder, where grib files are
+                or should be stored
+            delete_raw_data (bool,optional): select if downloaded raw data in
+                .grib.bz2 file format should be stored on the computer or
+                removed
+        """
+        self.clear()
+        # download files, if they don't already exist
+        file_names = download_icon_grib(run_date,
+                                        model_name=model_name,
+                                        download_dir=grib_dir)
+
+        # create centroids
+        nc_centroids_file = download_icon_centroids_file(model_name, grib_dir)
+        self.centroids = self._centroids_from_nc(nc_centroids_file)
+
+        # read intensity from files
+        for ind_i, file_i in enumerate(file_names):
+            gripfile_path_i = os.path.join(file_i[:-4])
+            with open(file_i, 'rb') as source, open(gripfile_path_i, 'wb') as dest:
+                dest.write(bz2.decompress(source.read()))
```

Comment:
Not very important, but I'd suggest moving the file decompression into the utils, and decompressing when you download (like you do with the centroids).

Reply:
I would prefer leaving it there. When delete_raw_data is set to False, and this function is run every day, an archive is build automatically, and there it makes sense to keep the compressed version.

Reply:
i'm fine with that 👍